Compute updates

anuchandy opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Please add an informative description that covers that changes made by the pull request.

If you are regenerating your SDK based off of a new swagger spec, please add the link to the corresponding swagger spec pull request that has been merged in the azure-rest-api-specs repository
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

anuchandy committed:
Regenerating the compute interfaces, implementing ListbyRegionAsync(..) and adding more async actions

anuchandy committed:
Regenerating disk, snapshot and scalesets collection interfaces to have async action methods

anuchandy committed:
Adding support for VM Boot Diagnostics, Adding tests that covers VM Boot diagnostics

anuchandy committed:
Adding VM Boot Diagnostics test recordings

anuchandy committed:
Expose Manager.RestClient and RestClient.Environment

anuchandy committed:
Ensure VM::disk's storage suffix is derived from the currently selected azure environment

anuchandy committed:
Recording for dynamic VM::disk storage suffix test

msftclas commented:
@anuchandy,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

anuchandy commented:
This PR mainly includes:

0. Exposing Manager::RestClient and RestClient::Environment (sometime impls needs to know the current azure environment)
1. Regenerated fluent compute interfaces
2. Added support for boot diagnostics
3. Added support to derive VM::disks storage uri suffix from the azure environment
4. Exposing action async methods in the fluent collection

anuchandy committed:
[1]. After extensions CUD refresh extensions (from VM model) that is exposed to user [2]. (Only for C#) Special handling for VM extension public and protected settings in Newtonsoft.Json.JObject format [3]. Test for [2]

anuchandy committed:
Test recordings for VM extensions

anuchandy committed:
Addresses the issue https://github.com/Azure/azure-sdk-for-net/issues/3398 [Unable to update VirtualMachineProfile Extension Public Setting (Scale Set)]

anuchandy committed:
Test recording for VMSS extensions

anuchandy committed:
Updating sample to pick custom script from updated location

anuchandy committed:
Re-recording the VM extension samples since we make additonal VM GET to fetch one with updated extensions list

anuchandy committed:
Use ConfigureAwait(false) whenever a sync method wraps corrosponding async method

anuchandy committed:
Use ConfigureAwait(false) in DiskImpl

anuchandy committed:
In playback mode, skip the part of extension test that validate storage blob uri created using storage data plane, doing so because we are not recording data plane api calls
